### PR TITLE
mcu: scale images

### PR DESCRIPTION
**Pull request by tronical**
Does this look like the right direction?

### CODE REVIEW

**Comment by ogoffart** [comment]
I guess this is somehow the right direction. 
We could in some cases know the size because if the with and height property is constant.  I think that might be the case in the printerdemo.
Also we don't want to render images bigger than they are on disk.

File: internal/core/graphics/image.rs [hunk 1]
```diff
@@ -238,6 +238,8 @@ pub enum ImageInner {
         /// The total size of the image (this might not be the size of the full image
         /// as some transparent part are not part of any texture)
         size: IntSize,
+        /// The size of the image before the compiler applied any scaling
+        unscaled_size: IntSize,
```
[the remark this answers is not in the document]
Comment (tronical) [reply internal/core/graphics/image.rs:242]: Agreed

**Comment by tronical** [comment]
Very good points. Thanks!

File: internal/compiler/passes/embed_images.rs [hunk 2]
```diff
@@ -227,7 +233,15 @@ fn convert_image(image: image::RgbaImage, format: PixelFormat, rect: Rect) -> Ve
...
-fn load_image(file: crate::fileaccess::VirtualFile) -> image::ImageResult<image::RgbaImage> {
+fn load_image(
+    file: crate::fileaccess::VirtualFile,
+) -> image::ImageResult<(image::RgbaImage, Size)> {
+    let scale_factor = std::env::var("SLINT_SCALE_FACTOR")
+        .ok()
+        .and_then(|x| x.parse::<f64>().ok())
+        .filter(|f| *f > 0.)
+        .unwrap_or(1.);
+
```
[the remark this answers is not in the document]
Comment (tronical) [reply internal/compiler/passes/embed_images.rs]: Done

**Review by ogoffart — APPROVED** [review]

File: internal/compiler/passes/embed_images.rs [hunk 3]
```diff
@@ -244,28 +274,55 @@ fn load_image(file: crate::fileaccess::VirtualFile) -> image::ImageResult<image:
             ))
         })?;
         // TODO: ideally we should find the size used for that `Image`
-        let size = tree.svg_node().size;
-        let mut buffer = vec![0u8; size.width() as usize * size.height() as usize * 4];
+        let original_size = tree.svg_node().size;
+        let mut width = original_size.width();
+        let mut height = original_size.height();
+        if scale_factor < 1. {
```
[the remark this answers is not in the document]
Comment (tronical) [reply internal/compiler/passes/embed_images.rs:280]: You're right :)

**Pull request merged**